### PR TITLE
Add support for local rest and sql runtimes

### PR DESCRIPTION
**Pull request by kellymtrinh**
## Description
Pulls in https://github.com/airplanedev/lib/pull/302, which implements the latest rest and sql runtimes for local execution.
Starts passing in the attached config variables for SQL and REST tasks.

- [x] merge and release and use version from lib PR first

## Test plan

- Implements the local SQL and REST runtimes so users can run SQL/REST tasks using `airplane dev` or in the editor. The main logic is in each type's `PrepareRun` function, used in CLI. 

- The `pkg/expressions` folder is pulled from airport (minus the v8go part, which requires CGO 😓 ). I will followup with a PR to pull this into airport. This is to support the common sql/rest use cases of using params or configs in JSTs: `value: {{params.my_value}}` and `{{configs.API_KEY}}` . 

## Test plan

 <details><summary> test child REST and SQL tasks from `airplane.execute()` 
</summary>

Screen Shot 2022-09-27 at 3 53 01 PM
(I checked that the return values of the sql and rest subtasks was expected)
</details>

<details><summary> test REST task execution works, with both config and param JSTs </summary>
Rest yaml, that uses both params and config JSTS: 
Rest yaml:

```
name: rest test
slug: rest_test
parameters:
  - name: User name
    slug: user_name
    type: shorttext
    description: Username
  - name: file name
    slug: my_file
    type: shorttext
    description: file
rest:
  resource: my_rest
  method: GET
  urlParams:
    email: "{{params.user_name}}"
    file: "{{params.my_file}}"
  path: /posts/1
  bodyType: json
  body: "{{params.my_file}}"
  formData:
    file: "{{params.my_file}}"
  headers:
    X-API-KEY: "{{configs.API_KEY}}"
    X-API-KEY-2: "{{params.my_file}}"
  configs:
    - API_KEY
```

I verified the param values are correct, and the config comes from the config set in the `dev.yaml`:
Screen Shot 2022-09-27 at 3 57 22 PM

</details>

<details><summary>test SQL execution, with JST queryArgs</summary>

Screen Shot 2022-09-27 at 3 49 40 PM

SQL yaml:
```
name: test sql
slug: test_sql
description: Selects all runs from a team with x limit. Team ID example is 1jq9NDeTcTmyqqXAjEsk7V0wXaV
parameters:
- name: limit
  slug: limit
  type: float
- name: team id
  slug: name
  type: shorttext
sql:
  resource: my_db
  entrypoint: test_sql.sql
  queryArgs:
    user_limit: "{{params.limit}}"
    team_id: "{{params.name}}"
  configs:
  - test_me
```

sql file:
```
select * from runs where team_id=:team_id limit :user_limit;
```

</details>

 <details><summary> test regular non editor `airplane dev` for REST task and SQL task:
</summary>

Screen Shot 2022-09-27 at 4 21 07 PM
Screen Shot 2022-09-27 at 4 21 50 PM

</details>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,7 +5,7 @@ go 1.19
 require (
 	github.com/AlecAivazis/survey/v2 v2.3.6
 	github.com/MakeNowJust/heredoc v1.0.0
-	github.com/airplanedev/lib v0.5.77
```
Comment (kellymtrinh) [inline go.mod:8]: Note to self to use the merged/released version of CLI in https://github.com/airplanedev/lib/pull/302

File: pkg/conf/dev.go [hunk 2]
```diff
@@ -97,6 +98,15 @@ func (d *DevConfig) RemoveResource(slug string) error {
 	return nil
 }
 
+// GetAttachedConfigs returns the configs that are attached to a task
+func GetAttachedConfigs(attatched []api.ConfigAttachment, allConfigs map[string]string) map[string]string {
```
Comment (calufornia) [inline pkg/conf/dev.go:102]:
```suggestion
func GetAttachedConfigs(attached []api.ConfigAttachment, allConfigs map[string]string) map[string]string {
```

Comment (calufornia) [inline pkg/conf/dev.go:102]: I might move this file to the `configs` package! It's kind of similar to `GenerateAliasToResourceMap` for resources, and so it technically can be used outside the context of the dev config file.

Comment (calufornia) [inline pkg/conf/dev.go:102]: Maybe call it something like `MaterializeConfigs`? Since `GetAttachedConfigs` sounds similar to `GetConfigAttachments`

**Review by calufornia — APPROVED** [review]
🎉

**Pull request merged**